aws/resource_aws_api_gateway_documentation_part_test.go

```diff
@@ -0,0 +1,243 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSAPIGatewayDocumentationPart_basic(t *testing.T) {
+	var conf apigateway.DocumentationPart
+
+	rString := acctest.RandString(8)
+	apiName := fmt.Sprintf("tf_acc_api_%s", rString)
+	properties := `{\"description\":\"Terraform Acceptance Test\"}`
+	uProperties := `{\"description\":\"Terraform Acceptance Test Updated\"}`
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSAPIGatewayDocumentationPartDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSAPIGatewayDocumentationPartConfig(apiName, properties),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSAPIGatewayDocumentationPartExists("aws_api_gateway_documentation_part.test", &conf),
```

Comment:
Nitpick (personal): I've been finding it helpful to declare the resource names as a top level variable in the function so its easier to duplicate code across functions/resources. e.g. `resourceName := "aws_api_gateway_documentation_part.test"` - personal opinion though!

Reply:
I'm ok with that, sounds reasonable, the only thing is that the majority of the tests we have in the codebase now doesn't follow this, so we might want to revisit the existing code, so other folks copy-pasting code know what's "preferred".